mdformat_tables/plugin.py

```diff
@@ -12,26 +12,18 @@ def update_mdit(mdit: MarkdownIt) -> None:
 
 
 def _parse_cells(
-    rows: List[List[List[Token]]], renderer: MDRenderer, options: dict, env: dict
+    rows: List[List[RenderTreeNode]],
+    renderer_funcs: Mapping[str, RendererFunc],
+    options: Mapping[str, Any],
+    env: MutableMapping,
 ) -> List[List[str]]:
-    """Convert tokens in each cell to strings."""
-    for i, row in enumerate(rows):
-        for j, cell_tokens in enumerate(row):
-            rows[i][j] = (
-                renderer.render(
-                    [Token("paragraph_open", "p", 1)] + cell_tokens
-                    or [Token("text", "", 0)] + [Token("paragraph_close", "p", -1)],
-                    options,
-                    env,
-                    finalize=False,
-                )
-                .replace(MARKERS.BLOCK_SEPARATOR, "")
-                .rstrip()
-            )
-    return rows
+    """Convert nodes in each cell to strings."""
+    return [[cell.render(renderer_funcs, options, env) for cell in row] for row in rows]
```

Comment:
I assume there is no longer the issue that inline text needs to be in a paragraph to be render correctly

Reply:
Hmm, not sure but I would my gut says there is no such Markdown text that would parse into inline text outside a paragraph. If there is a case we should add a test.

Also the code that adds a surrounding paragraph was buggy, (line 23 in the red diff is dead code) yet I never bumped into any issues, so I assume that is not needed.

Reply:
> my gut says there is no such Markdown text that would parse into inline text outside a paragraph.

Thats exactly how text in table cells work; they are not enclosed in a paragraph, just `th` or `td` tags

Reply:
Ah ok, but tests pass so :man_shrugging: 

Seems to work! :smile:

Reply:
Oh btw, this code passes all GFM spec tests in https://github.com/hukkinj1/mdformat-gfm/pull/2

So I'm very confident.

Reply:
Yeh if the tests pass then should be fine 👍 it was just to note this change in behaviour